code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -130,6 +131,141 @@
 				A.ex_act(severity)
 		qdel(src)
 
+
+
+
+
+
+//Table interactions
+/obj/structure/closet/crate/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if (istype(mover, /obj/structure/closet/crate))//Handle interaction with other crates
+		var/obj/structure/closet/crate/C = mover
+		if (tablestatus == 1 && C.tablestatus != 1 && !C.opened)//Allow the crate to slide under us if we're ontop of a table
+			return 1
+		else if (tablestatus == -1 && C.tablestatus == 1)//Allow it to slide over us if we're underneath a table
+			return 1
+		else//Otherwise, block it. Don't allow two crates on the same level of a tile
+			return 0
+	if(istype(mover,/obj/item/projectile))
+		if (tablestatus == 1)//They always block shots on a table
+			return 0
+		else if (!tablestatus && prob(15))//Usually will not block shots when lying on the floor
+			return 0
+		else return 1
+	else if(istype(mover) && mover.checkpass(PASSTABLE))
+		return 1
+	return 0
+
+/obj/structure/closet/crate/Move(var/turf/destination, dir)
+	if(..())
+		if (locate(/obj/structure/table) in destination)
+			if (tablestatus != 1)
+				set_tablestatus(-1)//Slide under the table
+		else
+			set_tablestatus(0)
+
+
+/obj/structure/closet/crate/toggle(var/mob/user)
+	if (!opened && tablestatus == -1)
+		user << span("warning", "You can't open that while it's under the table")
+		return 0
+	else
+		..()
+
+/obj/structure/closet/crate/proc/set_tablestatus(var/target)
+	if (tablestatus != target)
+		tablestatus = target
+
+	spawn(3)//Short spawn prevents things popping up where they shouldnt
+		switch (target)
+			if (1)
+				layer = LAYER_ABOVE_TABLE
+				pixel_y = 8
+			if (0)
+				layer = initial(layer)
+				pixel_y = 0
+			if (-1)
+				layer = LAYER_UNDER_TABLE
+				pixel_y = -4
+
+
+//For putting on tables
+/obj/structure/closet/crate/MouseDrop(atom/over_object)
+	if (istype(over_object, /obj/structure/table))
+		put_on_table(over_object, usr)
+		return 1
+	else
+		return ..()
+
+
+
+/obj/structure/closet/crate/proc/put_on_table(var/obj/structure/table/table, var/mob/usr)

```

Comment:
`var/mob/usr)` <-- nope. var/mob/user.

http://www.byond.com/docs/ref/info.html#/proc/var/usr

The fact that BYOND lets this compile is idiotic in and of itself, it should be a reserved word.